### ISSUE TITLE
Error occured in paperwm startup on RHEL 8.7, Gnome 3.32.2

### ISSUE THREAD

**Issue by fumbles** [paperwm/PaperWM#509]
**Describe the bug**
Error occured in paperwm startup on RHEL 8.7, Gnome 3.32.2
When I lock the laptop I'm hit with a bunch of errors from PaperWM

```
PaperWM
Error occured in paperwm startup:

this.get(...) is undefined

monitorsChanged@/home/daniel/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:1360:9
init@/home/daniel/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:1337:9
initWorkspaces@/home/daniel/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2235:13
```

**To Reproduce**
Lock screen..

**Expected behavior**
No errors

**Screenshots**
![image](https://user-images.githubusercontent.com/90968/230618668-238bf45a-070c-420e-b0f6-57001a60f297.png)

**System information:**
Please execute `./gather-system-info.sh` in you PaperWM clone and paste the output below.

```
Distribution: Red Hat Enterprise Linux
GNOME Shell 3.32.2
PaperWM branch/tag: develop
PaperWM commit: 9f3a5781b02d712454403dd65b841bf11b7478fe
Enabled extensions:
- appindicatorsupport@rgcjonas.gmail.com
- apps-menu@gnome-shell-extensions.gcampax.github.com
- dash-to-dock@gnome-shell-extensions.gcampax.github.com
- desktop-icons@gnome-shell-extensions.gcampax.github.com
- horizontal-workspaces@gnome-shell-extensions.gcampax.github.com
- launch-new-instance@gnome-shell-extensions.gcampax.github.com
- places-menu@gnome-shell-extensions.gcampax.github.com
- sasgui-status-icon@ibm.com
- top-icons@gnome-shell-extensions.gcampax.github.com
- updates-dialog@gnome-shell-extensions.gcampax.github.com
- user-theme@gnome-shell-extensions.gcampax.github.com
- window-list@gnome-shell-extensions.gcampax.github.com
```

Had to add in `ls /usr/share/gnome-shell/extensions/` to list extensions
**Additional context**

**Comment by Lythenas** [comment]
For older gnome versions please use the corresponding branch. From the readme:

> 3.28-3.38: https://github.com/paperwm/PaperWM/releases/tag/38.2

**Comment by fumbles** [comment]
I used that branch. 

It doesn't include the gather information script hence I used the dev branch to gather the info.

**Comment by Lythenas** [comment]
Ah sorry about that, maybe we need to push that script to the other branches too...

**Comment by Lythenas** [comment]
Hmm, actually this code hasn't changed since 38.2:

```
        this.monitors = new Map();
        this.get(workspaceManager.get_active_workspace()).getWindows().forEach(w => {
            animateWindow(w);
        });
```

So I'm a bit surprised about this error. Do the others have any idea?

**Comment by fumbles** [comment]
> Ah sorry about that, maybe we need to push that script to the other branches too...

No worries... I initially had used the wrong branch before reading closely that my level was within that range.  I could've specified I used that release in the top.

**Comment by jtaala** [comment]
Hmmm - sounds pretty similar to issues that #477  fixes (namely FIX 2 in that PR).  Especially given the repeated outputs and the lockscreen relation.

PaperWM has changed quite a bit since that release, but for testing I could rebase (or just cherry-pick the relevant commits) that PR on top of 38.2 branch if you're able to test it for us @fumbles?

**Comment by jtaala** [comment]
I could also throw in the `gather-system-info` script into the (rebased) PR for for 38.2 as well.

Oh @Lythenas - did you see my comment on #485? (re added output if wayland/x11 - love this script btw).

**Comment by fumbles** [comment]
@jtaala I can test it.
I'm probably going to ask our IT folks while we're using that version of gnome-shell also.

**Comment by jtaala** [comment]
Cool - okay, I've pushed a branch for this testing.  Please do a

```
git fetch --all
```

and  then checkout the branch:

```
git checkout gnome-3.38-fixes-deprecatedActorCall-guardAgainstMultipleDisables
```

and then do a logout/login and see if this helps at all.

> I'm probably going to ask our IT folks while we're using that version of gnome-shell also.

Yeah, it'll be a RHEL thing - their shell they use is based off an old gnome-shell version.  Same thing at my work (although I don't use gnome at my work, since we have virtually/remote accessed dev machines - that has other issues with gnome).

**Comment by fumbles** [comment]
Cool. I'll try that now.

Heads up:

```./gather-system-info.sh
Please include this information in your bug report on GitHub!
Distribution: Red Hat Enterprise Linux
GNOME Shell 3.32.2
PaperWM branch/tag: gnome-3.38-fixes-deprecatedActorCall-guardAgainstMultipleDisables
PaperWM commit: cb5c05c1f496ef216a426be424c74bb9b64584f6
Enabled extensions:
./gather-system-info.sh: line 30: gnome-extensions: command not found <-- I used ls to list the extensions earlier
```

**Comment by jtaala** [comment]
> Cool. I'll try that now.
> 
> Heads up:
> 
> ```shell
> Please include this information in your bug report on GitHub!
> Distribution: Red Hat Enterprise Linux
> GNOME Shell 3.32.2
> PaperWM branch/tag: gnome-3.38-fixes-deprecatedActorCall-guardAgainstMultipleDisables
> PaperWM commit: cb5c05c1f496ef216a426be424c74bb9b64584f6
> Enabled extensions:
> ./gather-system-info.sh: line 30: gnome-extensions: command not found <-- I used ls to list the extensions earlier
> ```

Oh weird, I just copied the `gather-system-info.sh` file from normal dev branch - I guess things have changed since 3.38... although, how did you run that script previously (when creating this commit?).

**Comment by jtaala** [comment]
> ```shell
> ./gather-system-info.sh: line 30: gnome-extensions: command not found <-- I used ls to list the extensions earlier
> ```

Oh, sorry I see - missed reading that line properly.

All good, (@Lythenas) have noted that for 3.38 would need to find an alternative to `gnome-extensions` for listing extensions if we want to add the script to the `gnome-3.38` branch.

**Comment by fumbles** [comment]
So far so good- no errors after locking screen at all.
 Last time I had a high gpu warning after a while of using paperwm intensively so we'll see how this goes :grimacing:

**Comment by jtaala** [comment]
> So far so good- no errors after locking screen at all. Last time I had a high gpu warning after a while of using paperwm intensively so we'll see how this goes grimacing

Yeah, give it a good workout.  Good test will be to do the following (in this order):

1. disable PaperWM (in extensions app or whatever);
2. enable PaperWM
3. lock screen
4. unlock screen

See #477 FIX 2 for checking the logs after unlocking (if needed or if there's still issues to report).

**Comment by fumbles** [comment]
> > So far so good- no errors after locking screen at all. Last time I had a high gpu warning after a while of using paperwm intensively so we'll see how this goes grimacing
> 
> Yeah, give it a good workout. Good test will be to do the following (in this order):
> 
> 1. disable PaperWM (in extensions app or whatever);
> 2. enable PaperWM
> 3. lock screen
> 4. unlock screen
> 
> See #477 FIX 2 for checking the logs after unlocking (if needed or if there's still issues to report).

Appreciate it...still going strong. :crossed_fingers: 

```Apr 07 16:16:34 d-rhel8 gnome-shell[3604]: loading default theme (Adwaita)
Apr 07 16:16:34 d-rhel8 gnome-shell[3604]: #paperwm enable 1680902024251#
Apr 07 16:16:34 d-rhel8 gnome-shell[3604]: enable called without calling disable

```

I did have one time where all extensions were disabled so i rebooted.

**Comment by jtaala** [comment]
> I did have one time where all extensions were disabled so i rebooted.

Hmm, okay, if happens again - see if we can reproduce this (sometimes gnome or extensions, paperWM or others can cause this on login - although I haven't had it for a while on login and not from unlocking).

**Comment by Lythenas** [comment]
I also noticed that `38.2` is actually a tag and not a branch. So we probably need to create a new branch for fixes that are gnome-shell 3.38 specific.

**Comment by jtaala** [comment]
> I also noticed that `38.2` is actually a tag and not a branch. So we probably need to create a new branch for fixes that are gnome-shell 3.38 specific.

Oh, yes!  I thought the `gnome-3.38` branch was what we told people to use (not tag `38.2`).  Should I just set `gnome-3.38` to where `38.2` is? (assuming `38.2` is ahead of `gnome-3.38` branch?).

**Comment by Lythenas** [comment]
Ah, I totally overlooked that branch :D I just saw that the readme points to the `38.2` tag. It looks like the tag is also newer than the branch. So yes I think we can set the branch to the tag, so we can continue development there.

**Comment by jtaala** [comment]
> Ah, I totally overlooked that branch :D I just saw that the readme points to the `38.2` tag. It looks like the tag is also newer than the branch. So yes I think we can set the branch to the tag, so we can continue development there.

done.

**Comment by jtaala** [comment]
@fumbles, @Lythenas and I just noticed that the `gnome-3.38` branch was behind the normally recommended `38.2` tag.  Anyways, I've also updated the `gnome-3.38-fixes-deprecatedActorCall-guardAgainstMultipleDisables` branch.

If you're on that branch (that you're testing), do a:

```
git pull
```

When you can.

P.S. is that branch still working okay?

**Comment by fumbles** [comment]
@Lythenas @jtaala 

Nice! I'll pull and log out/back in. 

So far everything has been working great. No issues since. 
No issues logging out or locking the laptop. No subsequent warnings about gpu usage. Been steady. Anything weird has been user error from still learning the key shortcuts.

**Comment by jtaala** [comment]
Sounds good.  Note that branch is old(er) and doesn't have the latest features, like easier winprop setting, focus-mode, window position indicators, and bunch of other features/fixes - so if you're even on `Gnome 43` check em out :+1: 

Jay.

**Comment by fumbles** [comment]
Closing this on out for y'all. I'm on fedora 37 now with the develop branch. But the `gnome-3.38-fixes-deprecatedActorCall-guardAgainstMultipleDisables` fixed my issues for RHEL 8.7

**Comment by jtaala** [comment]
> Closing this on out for y'all. I'm on fedora 37 now with the develop branch. But the `gnome-3.38-fixes-deprecatedActorCall-guardAgainstMultipleDisables` fixed my issues for RHEL 8.7

Cool, I'll merge that branch into the gnome-3.38 branch as well for others too.
